src/wallet/wallet.cpp

```diff
@@ -442,30 +442,40 @@ bool CWallet::Verify()
     if (GetBoolArg("-disablewallet", DEFAULT_DISABLE_WALLET))
         return true;
 
+    SoftSetArg("-wallet", DEFAULT_WALLET_DAT);
+
     uiInterface.InitMessage(_("Verifying wallet..."));
-    std::string walletFile = GetArg("-wallet", DEFAULT_WALLET_DAT);
 
-    std::string strError;
-    if (!CWalletDB::VerifyEnvironment(walletFile, GetDataDir().string(), strError))
-        return InitError(strError);
+    for (const std::string& walletFile : mapMultiArgs.at("-wallet")) {
+        if (walletFile.find_first_of("/\\") != std::string::npos) {
```

Comment:
`walletFile.find_first_of("/\\")` is not portable. Likely, boost::filesystem::path has a way to check a path for path separators.